File: src/features/contact-inquiries/dto/confirm-contact-inquiry.dto.ts
import { Transform } from 'class-transformer'
import { IsEmail, IsString, Matches } from 'class-validator'

export class ConfirmContactInquiryDto {
  @IsEmail()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim().toLowerCase() : value))
  email: string

  @IsString()
  @Matches(/^\d{6}$/, { message: 'Verification code must be 6 digits.' })
  code: string
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/contactinquiries/ConfirmContactInquiryDto.java
package com.mytegroup.api.dto.contactinquiries;

import com.mytegroup.api.dto.common.ValidationConstants;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record ConfirmContactInquiryDto(
    @NotBlank(message = "Email is required")
    @Email(message = "Email must be valid")
    String email,
    @NotBlank(message = "Code is required")
    @Pattern(regexp = ValidationConstants.VERIFICATION_CODE_REGEX, message = "Code must be 6 digits")
    String code
) {
    public ConfirmContactInquiryDto {
        if (email != null) {
            email = email.toLowerCase().trim();
        }
    }
}

